### PR TITLE
HBASE-18382 add transport type info into Thrift UI

### PR DESCRIPTION
**Pull request by bea0113**
Changed UI on info server to be more informative. 
Added QOP settings info, changed protocol info from true/false to Compact/Binary, changed transport info from true/false to Framed/Standard.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 44s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 42s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 58s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m  0s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 13s |  The patch does not generate ASF License warnings.  |
|  |   |  20m 13s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/880 |
| Optional Tests | dupname asflicense javac javadoc unit |
| uname | Linux 90887c4c7f82 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-880/out/precommit/personality/provided.sh |
| git revision | master / ea6cea846a |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/1/testReport/ |
| Max. process+thread count | 1738 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp [hunk 1]
```diff
@@ -31,9 +31,12 @@ long startcode = conf.getLong("startcode", System.currentTimeMillis());
 String listenPort = conf.get("hbase.regionserver.thrift.port", "9090");
 String serverInfo = listenPort + "," + String.valueOf(startcode);
 ImplType implType = ImplType.getServerImpl(conf);
-String framed = implType.isAlwaysFramed()
-    ? "true" : conf.get("hbase.regionserver.thrift.framed", "false");
-String compact = conf.get("hbase.regionserver.thrift.compact", "false");
+String framed = (implType.isAlwaysFramed()
+    ? "true" : conf.get("hbase.regionserver.thrift.framed", "false"))
```
Comment (bharathv) [inline hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp]: "true" -> "Framed"?

nit: I think the logic has become nested and is unreadable for a ternary operator. Mind switching to an if-else?

File: hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp [hunk 2]
```diff
@@ -31,9 +31,12 @@ long startcode = conf.getLong("startcode", System.currentTimeMillis());
...
+    .equals("true") ? "Framed": "Standard";
+String compact = conf.get("hbase.regionserver.thrift.compact", "false")
```
Comment (bharathv) [inline hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp]: nit: String protocol =...

Comment (bharathv) [reply hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp]: Also switch to conf.getBoolean() and avoid equals..?

File: hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp [hunk 3]
```diff
@@ -112,14 +115,19 @@ String compact = conf.get("hbase.regionserver.thrift.compact", "false");
...
-            <td>Framed Transport</td>
+            <td>Transport</td>
             <td><%= framed %></td>
-            <td>Thrift RPC engine uses framed transport</td>
+            <td>Thrift RPC engine transport type</td>
+        </tr>
+        <tr>
+            <td>Quality Of Protection</td>
+            <td><%= qop %></td>
+            <td>QOP settings for SASL </td>
```
Comment (bharathv) [inline hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp]: nit: extra space after SASL.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 12s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 57s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 34s |  master passed  |
| -0 :warning: |  patch  |   6m 42s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 31s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  19m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/880 |
| Optional Tests | dupname asflicense javac javadoc unit |
| uname | Linux 2b982123cc09 4.15.0-70-generic #79-Ubuntu SMP Tue Nov 12 10:36:11 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-880/out/precommit/personality/provided.sh |
| git revision | master / 0f166edc3d |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/2/testReport/ |
| Max. process+thread count | 1716 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 24s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 33s |  master passed  |
| -0 :warning: |  patch  |   7m  8s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   5m 24s |  root in the patch failed.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  javadoc  |   0m 15s |  hbase-thrift in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 22s |  hbase-thrift in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 11s |  The patch does not generate ASF License warnings.  |
|  |   |  15m 56s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/880 |
| Optional Tests | dupname asflicense javac javadoc unit |
| uname | Linux 9442040c6893 4.15.0-70-generic #79-Ubuntu SMP Tue Nov 12 10:36:11 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-880/out/precommit/personality/provided.sh |
| git revision | master / 0f166edc3d |
| Default Java | 1.8.0_181 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/3/artifact/out/patch-mvninstall-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/3/artifact/out/patch-javadoc-hbase-thrift.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/3/artifact/out/patch-unit-hbase-thrift.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/3/testReport/ |
| Max. process+thread count | 86 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 34s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 34s |  master passed  |
| -0 :warning: |  patch  |   6m 22s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 56s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 38s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  17m 13s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/880 |
| Optional Tests | dupname asflicense javac javadoc unit |
| uname | Linux 961057d6d83a 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-880/out/precommit/personality/provided.sh |
| git revision | master / 0f166edc3d |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/4/testReport/ |
| Max. process+thread count | 1863 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp [hunk 4]
```diff
@@ -31,9 +31,11 @@ String serverType = (String)getServletContext().getAttribute("hbase.thrift.serve
 long startcode = conf.getLong("startcode", System.currentTimeMillis());
 String listenPort = conf.get("hbase.regionserver.thrift.port", "9090");
 ImplType implType = ImplType.getServerImpl(conf);
-String framed = implType.isAlwaysFramed()
-    ? "true" : conf.get("hbase.regionserver.thrift.framed", "false");
-String compact = conf.get("hbase.regionserver.thrift.compact", "false");
+
+String transport = implType.isAlwaysFramed() ||conf.getBoolean("hbase.regionserver.thrift.framed", false) ? "Framed" : "Standard";
```
Comment (bharathv) [inline hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp]: Couple of nits:

- Ternary operator's boolean expression should be parenthesized, as per the Java style guide that HBase follows. Also a space after '||'. https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-136091.html
-  Can you please take care of max line width (100 chars). Looks like checkstyle does not run in jsp files. GOod to fix it before committing.

Comment (virajjasani) [reply hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp]: +1 to the suggestions, rest looks good

Comment (bea0113) [reply hbase-thrift/src/main/resources/hbase-webapps/thrift/thrift.jsp]: I corrected these, I hope it looks good now :)

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 18s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 56s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 33s |  master passed  |
| -0 :warning: |  patch  |   6m 41s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 35s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 33s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 41s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 13s |  The patch does not generate ASF License warnings.  |
|  |   |  22m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/880 |
| Optional Tests | dupname asflicense javac javadoc unit |
| uname | Linux 18aaa459d957 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-880/out/precommit/personality/provided.sh |
| git revision | master / dff7fffe85 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/5/testReport/ |
| Max. process+thread count | 1808 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-880/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**